pkg/deploy/api/types.go

```diff
@@ -328,3 +328,22 @@ type DeploymentConfigRollbackSpec struct {
 	// IncludeStrategy specifies whether to include the deployment Strategy.
 	IncludeStrategy bool
 }
+
+// DeploymentLogs represents the logs for a deployment
+type DeploymentLogs struct {
+	kapi.TypeMeta
+	kapi.ListMeta
+}
+
+// DeploymentLogOptions is the REST options for a deployment log
+type DeploymentLogOptions struct {
+	kapi.TypeMeta
+
+	// Follow if true indicates that the deployment log should be streamed until
+	// the deployment terminates.
+	Follow bool
+
+	// NoWait if true causes the call to return immediately even if the deployment
+	// is not available yet. Otherwise the server will wait until the deployment has started.
+	NoWait bool
```

Comment:
Note I'm not against trying to solve these, just that it's not required to
ship 3.1

On Wed, Jul 29, 2015 at 4:03 PM, Clayton Coleman <ccoleman@redhat.com>
wrote:

> I don't think in the short term there is a requirement that we use oc logs
> for all types of logs.  It's a nice to have.
>
> On Wed, Jul 29, 2015 at 10:43 AM, Michail Kargakis <
> notifications@github.com> wrote:
>
>> In pkg/deploy/api/types.go
>> <https://github.com/openshift/origin/pull/3943#discussion_r35767333>:
>>
>> > +type DeploymentLogs struct {
>> > +	kapi.TypeMeta
>> > +	kapi.ListMeta
>> > +}
>> > +
>> > +// DeploymentLogOptions is the REST options for a deployment log
>> > +type DeploymentLogOptions struct {
>> > +	kapi.TypeMeta
>> > +
>> > +	// Follow if true indicates that the deployment log should be streamed until
>> > +	// the deployment terminates.
>> > +	Follow bool
>> > +
>> > +	// NoWait if true causes the call to return immediately even if the deployment
>> > +	// is not available yet. Otherwise the server will wait until the deployment has started.
>> > +	NoWait bool
>>
>> If this doesn't go upstream we won't be able to specify it via oc logs
>> (when oc logs will support builds and deployments).
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/openshift/origin/pull/3943/files#r35767333>.
>>
>
>
>
> --
> Clayton Coleman | Lead Engineer, OpenShift
>



-- 
Clayton Coleman | Lead Engineer, OpenShift